crates/server/src/lib.rs

```diff
@@ -16,88 +16,140 @@ use actix_web::{
 use handlers::assets::handle_assets;
 use handlers::not_found::handle_not_found;
 use handlers::worker::handle_worker;
-use std::fs::OpenOptions;
-use std::path::Path;
-use std::sync::RwLock;
+use std::{
+    fs::{File, OpenOptions},
+    path::PathBuf,
+    sync::RwLock,
+};
 use wws_api_manage::config_manage_api_handlers;
 use wws_data_kv::KV;
 use wws_panel::config_panel_handlers;
-use wws_router::Routes;
+use wws_router::{Routes, WORKERS};
+
+#[derive(Clone, PartialEq)]
+pub enum Panel {
+    Enabled,
+    Disabled,
+}
+
+impl From<bool> for Panel {
+    fn from(panel_enabled: bool) -> Self {
+        if panel_enabled {
+            Panel::Enabled
+        } else {
+            Panel::Disabled
+        }
+    }
+}
 
 #[derive(Default)]
 pub(crate) struct DataConnectors {
     kv: KV,
 }
 
+#[derive(Clone)]
+pub struct ServeOptions {
+    pub root_path: PathBuf,
+    pub base_routes: Routes,
+    pub hostname: String,
+    pub port: u16,
+    pub panel: Panel,
+    pub stderr: Option<PathBuf>,
+    pub cors_origins: Option<Vec<String>>,
+}
+
+#[derive(Default)]
+pub struct AppData {
+    routes: Routes,
+    root_path: PathBuf,
+    stderr: Option<File>,
+    cors_origins: Option<Vec<String>>,
+}
+
+impl TryFrom<ServeOptions> for AppData {
+    type Error = ServeError;
+
+    fn try_from(serve_options: ServeOptions) -> Result<Self> {
+        let stderr = if let Some(stderr) = serve_options.stderr {
+            Some(
+                OpenOptions::new()
+                    .append(true)
+                    .open(stderr)
+                    .map_err(|_| ServeError::InitializeServerError)?,
+            )
+        } else {
+            None
+        };
+
+        Ok(AppData {
+            routes: serve_options.base_routes,
+            root_path: serve_options.root_path.clone(),
+            stderr,
+            cors_origins: serve_options.cors_origins.clone(),
+        })
+    }
+}
+
 /// Initializes an actix-web server based on the given configuration and
 /// path. It will configure the different handlers to manage static
 /// assets and workers.
-pub async fn serve(
-    root_path: &Path,
-    base_routes: Routes,
-    hostname: &str,
-    port: u16,
-    panel: bool,
-    stderr: Option<&Path>,
-    cors_origins: Option<Vec<String>>,
-) -> Result<Server> {
+pub async fn serve(serve_options: ServeOptions) -> Result<Server> {
```

Comment:
Some arguments are "mandatory" for the server initialization. I can think of `root_path`, `routes`, `hostname` and `port`. The other arguments are optional. What do you think about keeping them as independent arguments?

```suggestion
pub async fn serve(root: &Path, routes: Routes, hostname: &str, port: u16, options: ServeOptions) -> Result<Server> {
```

I'm thinking on other similar APIs, which sets mandatory arguments and then, an open `options` object for extra params.

Reply:
Updated the PR with all optional attributes. I prefer the struct option so that arguments are 'named', as opposed to positional heterogeneous arguments on the caller.

Reply:
Sounds good! I'm not very familiar with Rust conventions yet, so either options work for me! 👍 to merge it as a struct.